### PR TITLE
Add, use, and validate OpenAPI specification

### LINKED ISSUE #42: Write and use API specification
Component parts:
- [x] Create `openapi.yml`
- [x] Validate `openapi.yml` in the CI build
- [x] Add OpenAPI validation badge to the README
- [x] Add Committee gem and enable request validation (ideally, also add  response validation; if you don't, add the commented-out response validation code and write up a new issue), then fix all the stuff that breaks
- [x] Create `docs/` folder (copy artifacts from another repo, *e.g.*, dor-services-app
- [ ] [Enable GitHub Pages](https://github.com/sul-dlss/workflow-server-rails/settings) with the `docs/` folder as the source (and don't use the `Theme Chooser`)
- [ ] Verify that the automagic OpenAPI documentation works: https://sul-dlss.github.io/suri-rails/

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #42 

## Why was this change made?

To document the SURI API.

Includes:
* Draft initial API documentation
* Use Committee gem to validation requests and responses
* Add OpenAPI validation badge
* Validate OpenAPI in CI
* Add docs for GitHub Pages-based rendering of OpenAPI spec
* Pin Rubocop, generate new TODO config, and run auto-correct

## Was the documentation (README, DevOpsDocs, wiki, openapi.yml) updated?

Yes.

## Does this change affect how this application integrates with other services?

Nothing is using suri-rails in a deployed environment yet. I will test DSA with an updated suri-rails container based on this branch which I will create locally.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
   test:
     docker:
       # specify the version you desire here
-      - image: circleci/ruby:2.7-buster
+      - image: circleci/ruby:2.7.0-node
```
Comment (jcoyne) [inline .circleci/config.yml:16]: can you use 2.7.1, 2.7.0 has some security vulnerabilities.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - 'test/**/*'
+    - 'app/controllers/identifiers_controller.rb'
+    - 'app/models/application_record.rb'
+    - 'app/models/identifier.rb'
+    - 'config/application.rb'
+    - 'db/migrate/20170825145538_create_identifiers.rb'
+
+# Offense count: 1
+# Cop supports --auto-correct.
+Style/IfUnlessModifier:
```
Comment (jcoyne) [inline .rubocop_todo.yml:22]: Why did you introduce this issue? Can you fix rails_helper?

File: .circleci/config.yml (hunk 1 above)
Comment (mjgiarlo) [reply .circleci/config.yml]: done.

File: .rubocop_todo.yml (hunk 2 above)
Comment (mjgiarlo) [reply .rubocop_todo.yml]: Rubocop hijinks. Fixed.

**Comment by mjgiarlo** [comment]
@jcoyne take another look?

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -6,7 +6,8 @@
 require File.expand_path('../config/environment', __dir__)
 # Prevent database truncation if the environment is production
 abort('The Rails environment is running in production mode!') if Rails.env.production?
-require 'rspec/rails'
+
+equire 'rspec/rails'
```
Comment (jcoyne) [inline spec/rails_helper.rb:10]: typo

Comment (mjgiarlo) [reply spec/rails_helper.rb]: just pushed

**Pull request merged**